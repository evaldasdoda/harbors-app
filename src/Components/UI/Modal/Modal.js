import React from 'react';
import PropTypes from 'prop-types';

// Components
import Button from '../Button/Button';

// Styles
require('./Modal.scss');

export default class modal extends React.Component {
    constructor(props) {
        super(props);

        this.setWrapperRef = this.setWrapperRef.bind(this);
        this.handleClickOutside = this.handleClickOutside.bind(this);
    }

    componentDidMount() {
        if(this.props.backClick){
            document.addEventListener('mousedown', this.handleClickOutside);
        }
    }

    componentWillUnmount() {
        if(this.props.backClick){
            document.removeEventListener('mousedown', this.handleClickOutside);
        }
    }

    setWrapperRef(node) {
        if(this.props.backClick){
            this.wrapperRef = node;
        }
    }

    handleClickOutside(event) {
        if(this.props.backClick){
            if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
                this.props.onButtonClick();
            }
        }
    }
    render() {
        return (
            <div data-testid="modal" ref={this.setWrapperRef} className="MODAL">
                <div className="MODAL__container">
                    <h2>{this.props.header}</h2>
                    <hr />
                    <p>{this.props.content}</p>
                    <Button class="btn-danger" clicked={this.props.onButtonClick}>
                        Close
                    </Button>
                </div>
            </div>
        );
    }
}
modal.propTypes = {
    backClick: PropTypes.bool,
    header: PropTypes.string,
    content: PropTypes.string,
    onButtonClick: PropTypes.func,
};