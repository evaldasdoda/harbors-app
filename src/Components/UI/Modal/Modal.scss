@import "../../../Assets/style.scss";

.MODAL {
    &__container {
        top: 15%;
        width: 400px;
        left: calc(50% - 200px);
        position: fixed;
        background-color: $white;
        z-index: 999;
        color: $black;
        border-radius: $radius-sm;
        padding: $space-sm;
        box-shadow: $shadowelements;
        & button {
            float: right;
        }
    }
}
