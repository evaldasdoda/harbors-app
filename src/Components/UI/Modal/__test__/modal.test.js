import React from 'react';
import ReactDOM from 'react-dom';
import Modal from '../Modal';

import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup)

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Modal></Modal>, div)
})

it("renders modal correctly",()=>{
    const {getByTestId} = render(<Modal header="Header" content="Content" />)
    expect(getByTestId('modal'))
})


it("matches snapshot",()=>{
    const tree = renderer.create(
        <Modal
            header="Header"
            content="Content"
        />).toJSON();
    expect(tree).toMatchSnapshot();
})