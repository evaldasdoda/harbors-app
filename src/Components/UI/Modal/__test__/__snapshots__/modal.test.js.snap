// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MODAL"
  data-testid="modal"
>
  <div
    className="MODAL__container"
  >
    <h2>
      Header
    </h2>
    <hr />
    <p>
      Content
    </p>
    <button
      className="BUTTON btn btn-danger"
      data-testid="button"
    >
      Close
    </button>
  </div>
</div>
`;
